### PR TITLE
feat: added resource groups ids output

### LINKED ISSUE #184: feat: module should return resource groups created by the module
- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Description

When creating vnets for the spokes, the module output the ids of the vnets created but it does not expose the resource groups where those vnets were created.

This would be useful in cases like our where we need this resource groups to assign permissions to some service principals to handle those.

### Describe the solution you'd like
make the module to output those resource groups. something like 

```
output "resource_group_ids" {
  description = "The created resource_groups IDs, expressed as a map."
  value = {
    for k, v in azapi_resource. rg : k => v.id
  }
}
```

### PR DESCRIPTION
**Pull request by ppodevlab**
# Overview/summary
Module now returns a list of resource groups created by the virtualnetwork submodule.

## This PR fixes/adds/changes/removes
Fixes #184 

## As part of this pull request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-lz-vending/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-lz-vending/issues), for tracking and closure.
- [X] Run and `make fmt` & `make docs` to format your code and update documentation.
- [ ] Created unit and deployment tests and provided evidence.
- [X] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by ppodevlab** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

@microsoft-github-policy-service agree company="Enreach"

**Comment by matt-FFFFFF** [comment]
Hey @ppodevlab 

Thanks for your PR. I'll review it shortly.

File: outputs.tf [hunk 1]
```diff
@@ -8,12 +8,16 @@ output "subscription_resource_id" {
   description = "The subscription_resource_id is the Azure subscription resource id that resources have been deployed into"
 }
 
-
 output "virtual_network_resource_ids" {
   value       = local.virtual_network_resource_ids
   description = "A map of virtual network resource ids, keyed by the var.virtual_networks input map. Only populated if the virtualnetwork submodule is enabled."
 }
 
+output "resource_group_ids" {
```
Comment (matt-FFFFFF) [inline outputs.tf:16]: please can you rename this to:

```suggestion
output "virtual_network_resource_group_ids" {
```

This means we can add other outputs for future submodules that may create RGs

File: locals.tf [hunk 2]
```diff
@@ -25,4 +25,8 @@ locals {
   # virtual_networks_merged is a map of virtual networks created, if the module has been enabled.
   # This is used in the outputs.tf file to return the virtual network resource ids.
   virtual_network_resource_ids = try(module.virtualnetwork[0].virtual_network_resource_ids, {})
+
+  # resource_group_ids is a map of resource groups created, if the module has been enabled.
+  # This is used in the outputs.tf file to return the resource group ids.
+  resource_group_ids = try(module.virtualnetwork[0].resource_group_ids, {})
```
Comment (matt-FFFFFF) [inline locals.tf:31]: Please can we rename this as per other comments

```suggestion
  virtual_network_resource_group_ids = try(module.virtualnetwork[0].resource_group_ids, {})
```

File: outputs.tf [hunk 3]
```diff
@@ -8,12 +8,16 @@ output "subscription_resource_id" {
...
+  value       = local.resource_group_ids
```
Comment (matt-FFFFFF) [inline outputs.tf:17]:
```suggestion
  value       = local.virtual_network_resource_group_ids
```

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]
Hi, looks great just some minor changes please

**Comment by ppodevlab** [comment]
> Hi, looks great just some minor changes please

done

**Comment by matt-FFFFFF** [comment]
Will be in next minor with #187

**Pull request merged**